### PR TITLE
Select video in see the fishway

### PR DESCRIPTION
**Pull request by fungjj92**
## Overview

Build out See the Fishway. Videos can be selected and autoplayed in the sidebar with accompanying information.

Connects #36 

### Demo
Screen Shot 2019-05-14 at 2 54 56 PM

Styling super rough, obviously.

### Notes

I made a followup card to format the datetime, which was way more of a pain than I expected. Couldn't get Moment.js or Date.js working! 👎 #71 

## Testing Instructions
Netlify, visit See the Fishway tab.

Videos should autoplay. Custom controls to come.

### CODE REVIEW

File: src/app/src/components/VideoCard.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import styled from 'styled-components';
+import { themeGet } from 'styled-system';
+
+const StyledVideoCard = styled('img')(props => ({
+    borderRadius: '100px',
+    border: props.selected
+        ? `${themeGet('colors.greens.1')(props)} 3px solid`
+        : 'none',
```
Comment (maurizi) [inline src/app/src/components/VideoCard.js:9]: :tada: nice use of `styled-components` here.

File: src/app/src/components/Sidebar.js [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
+    background: ${themeGet('colors.teals.5')};
+    text-align: center;
+    width: 100%;
 `;
 
-const Sidebar = () => {
+const Sidebar = ({ fish }) => {
+    const { commonName, scientificName, path, notes, timestamp } = fish;
+    const date = new Date(timestamp).toLocaleString();
+    const video = fish && <video src={`${path}.mp4`} autoPlay={true} />;
```
Comment (maurizi) [inline src/app/src/components/Sidebar.js]: We need to add calls to `pauseTimer()` and `resetTimer()` here similar to in `AboutSlide`: https://github.com/azavea/pwd-fishway/blob/33567962c7826817849a271da692d7c10cfb29a5/src/app/src/components/AboutSlide.js#L84-L86

A nice bonus refactor on top of that you could consider is making a very small `Video` component that wires in these actions and passes the rest of its props to the native `<video>` element, and then using that here and in `AboutSlide`:
```jsx
const Video = (props) => {
   return <video
        onPlay={() => dispatch(pauseTimer())}
        onPause={() => dispatch(resetTimer())}
        onEnded={() => dispatch(resetTimer())} 
       {..props}
    />
};
```

**Review by maurizi — CHANGES_REQUESTED** [review]
Other than the comment re: calling the pause/reset timer for the screensaver this looks great.

File: src/app/src/components/Sidebar.js (hunk 2 above)
Comment (fungjj92) [reply src/app/src/components/Sidebar.js]: This is a great suggestion!! Good use of the spread operator too. Tackled in latest commit

**Comment by fungjj92** [comment]
I have to address a bug with the About video's click behavior.

File: src/app/src/components/AboutSlide.js [hunk 3]
```diff
@@ -48,6 +48,9 @@ export default class AboutSlide extends Component {
             this.autoplay();
         }
     }
+    setVideoRef(ref) {
+        this.videoRef = ref;
+    }
```
Comment (maurizi) [inline src/app/src/components/AboutSlide.js]: What is the purpose of this function? It doesn't seem to be called anywhere...

**Review by maurizi — APPROVED** [review]
Other than the 1 comment I left about what appears to be unused code, this is looking great!

Feel free to merge w/o further review after addressing the comment.

File: src/app/src/components/AboutSlide.js (hunk 3 above)
Comment (fungjj92) [reply src/app/src/components/AboutSlide.js]: Good catch, remnant of something I was trying out.

**Comment by fungjj92** [comment]
Thanks for review! Merging.

**Pull request merged**